share-extension/ShareViewController.m

```diff
@@ -868,5 +879,102 @@ - (NSString*)stringEncodedWithBase64:(NSString*)str
+                
+                // 2. Change property orientation to Normal
+                [metadataAsMutable setValue:[NSNumber numberWithInt:kCGImagePropertyOrientationUp] forKey:(NSString *)kCGImagePropertyOrientation];
+                [tiffMutableDic setValue:[NSNumber numberWithInt:kCGImagePropertyOrientationUp] forKey:(NSString *)kCGImagePropertyOrientation];
+                
+                [metadataAsMutable setValue:tiffMutableDic forKey:(NSString *)kCGImagePropertyTIFFDictionary];
+                
+                // 3. Creat a portrait photo from the provided photo.
+                NSData * photoData;
+                if (postActivity.url != nil){
+                    photoData = [NSData dataWithContentsOfURL:postActivity.url];
+                } else {
+                    photoData = postActivity.fileData;
+                }
+                UIImage * image = [UIImage imageWithData:photoData];
+                UIImage * img = [self resizeImage:image];
+                NSString * typeFile = (__bridge NSString *)(UTI);
+                if ([typeFile isEqualToString:@"pulic.png"]){
+                    photoData = UIImagePNGRepresentation(img);
+                    postActivity.fileExtension = @"png";
+                } else {
+                    photoData = UIImageJPEGRepresentation(img, kJPEGCompressionLevel);
+                    postActivity.fileExtension = @"jpg";
+                }
+                
+                //4. Assign the metadata to this new photo.
+                
+                CGImageSourceRef newPhotoSourceRef = CGImageSourceCreateWithData((CFDataRef)photoData,NULL);
+                //this will be the data CGImageDestinationRef will write into
+                NSMutableData * new_photoData = [NSMutableData data];
+                CGImageDestinationRef destination = CGImageDestinationCreateWithData((CFMutableDataRef)new_photoData,UTI,1,NULL);
+                
+                if(destination) {
+                    //add the image contained in the image source to the destination, overidding the old metadata with our modified metadata
+                    CGImageDestinationAddImageFromSource(destination,newPhotoSourceRef,0, (CFDictionaryRef) metadataAsMutable);
+                    
+                    //tell the destination to write the image data and metadata into our data object.
+                    //It will return false if something goes wrong
+                    BOOL success = NO;
+                    success = CGImageDestinationFinalize(destination);
+                    
+                    if(success) {
+                        postActivity.fileData = new_photoData;
+                        postActivity.url = nil;
+                    }
+                }
+                
+            }
+        }
+    });
+
+}
+- (UIImage *)resizeImage:(UIImage *)image {
```

Comment:
What does this method do exactly?
If this is were the image is rotated, the name should be `rotateImage`.